accepted/00000-remote-command-auditing.md

```diff
@@ -0,0 +1,106 @@
+- Feature Name: Remote Command Auditing
+- Start Date: 2019-06-21
+- RFC PR: TBD
+
+# Summary
+[summary]: #summary
+
+Implement a role-based access model to access `Remote Command` for managed systems and log every command executed.
+
+# Motivation
+[motivation]: #motivation
+
+At the time of the writing, there are three features to issue a remote command on a managed system or a group of systems registered to the product:
+- A `Remote Command` tab is present in the details section of a traditional or minion system (or a `System Set Manager` group)
+- The API exposes the `system.scheduleScriptRun` to issue a command
+- (Only for minions) a `Salt > Remote Commands` page to launch a command on minions by leveraging [Salt targeting](https://docs.saltstack.com/en/latest/topics/targeting/index.html) to identify the target(s)
+
+Each user can only target the systems within the same organization that the systems are registered to, provided that the user has a particular role:
+- Every user that has `SUSE Manager Administrator` or `Organization Administrator` role can access the tab and run a remote command on the selected system or a group of systems
+- The API is always exposed but checks for the role of the user supplying the command after the API is called
+- For every user registered to SUSE Manager, independent of the role, the menu item `Salt > Remote Commands` is accessible. Only users with the `SUSE Manager Administrator` or `Organization Administrator` role, though, can impact systems using this feature, whereas all other users cannot target any system (`"no systems found"`).
+
+Every system outside the organization of the user issuing the command is not impacted nor is it visible by the user.
+
+The motivations behind the changes introduced in this RFC are _security_ and _auditing_. Any command issues via the `Remote Command` will be run with `root` privileges on the target system(s). Additionally, the user that issued the command is not logged along with the command issues and its output.
```

Comment:
There is one extra twist worth noting if we frame this from the POV of security (and not just auditing): as soon as an admin can install an package, they can also run whatever root can via post scripts.

So if an admin can run commands, they can also install software. And if they can install software, they can also run commands.

Essentially, from a security point of view, it is either "you are root on system X" or "you are not" - we are not actually able to differentiate between "install" and "run command" (and, as @meaksh mentioned, even "configure"). 

Users have been vocal about auditing of the Remote Commands page just because the introduction of that page made it more visible, but from a security perspective the threat model hasn't really changed.

Reply:
Yes, I agree, I will add a paragraph describing the potential surface of attack when allowing every user to install packages.